azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/utils/MavenAuthUtils.java

```diff
@@ -0,0 +1,82 @@
+        "\"Azure\", \"AzureChina\", \"AzureGermany\", \"AzureUSGovernment\",.";
+
+    public static com.microsoft.azure.tools.auth.model.AuthConfiguration convertToAuthConfiguration(MavenAuthConfiguration auth)
+            throws InvalidConfigurationException {
+        if (Objects.isNull(auth)) {
+            return new com.microsoft.azure.tools.auth.model.AuthConfiguration();
+        }
+        final com.microsoft.azure.tools.auth.model.AuthConfiguration authConfiguration = new com.microsoft.azure.tools.auth.model.AuthConfiguration();
+        authConfiguration.setClient(auth.getClient());
+        authConfiguration.setTenant(auth.getTenant());
+        authConfiguration.setCertificate(auth.getCertificate());
+        authConfiguration.setCertificatePassword(auth.getCertificatePassword());
+        authConfiguration.setKey(auth.getKey());
+        authConfiguration.setHttpProxyHost(auth.getHttpProxyHost());
+        final String authTypeStr = auth.getType();
+        authConfiguration.setType(AuthType.parseAuthType(authTypeStr));
+        authConfiguration.setEnvironment(AuthHelper.stringToAzureEnvironment(auth.getEnvironment()));
+        if (Objects.nonNull(auth.getHttpProxyPort()) && !NumberUtils.isCreatable(auth.getHttpProxyPort())) {
+            throw new InvalidConfigurationException(String.format("Invalid integer number for httpProxyPort: '%s'", auth.getHttpProxyPort()));
+        }
+        authConfiguration.setHttpProxyPort(NumberUtils.toInt(auth.getHttpProxyPort()));
+
+        if (StringUtils.isNotBlank(auth.getEnvironment()) && Objects.isNull(authConfiguration.getEnvironment())) {
+            throw new InvalidConfigurationException(String.format(INVALID_AZURE_ENVIRONMENT, auth.getEnvironment()));
+        }
+        // if user specify 'auto', and there are SP configuration errors, it will fail back to other auth types
+        // if user doesn't specify any authType
+        if (StringUtils.isBlank(auth.getType())) {
+            if (!StringUtils.isAllBlank(auth.getCertificate(), auth.getKey(), auth.getCertificatePassword())) {
+                ValidationUtil.validateMavenAuthConfiguration(authConfiguration);
+            }
+        } else if (authConfiguration.getType() == AuthType.SERVICE_PRINCIPAL) {
+            ValidationUtil.validateMavenAuthConfiguration(authConfiguration);
+        }
+        return authConfiguration;
+    }
+
+    public static AzureCredentialWrapper login(MavenSession session, SettingsDecrypter settingsDecrypter, @Nonnull MavenAuthConfiguration auth)
+            throws AzureExecutionException, MavenDecryptException {
+        final String serverId = auth.getServerId();
+        final com.microsoft.azure.tools.auth.model.AuthConfiguration authConfiguration;
+        try {
+            authConfiguration = MavenAuthUtils.convertToAuthConfiguration(StringUtils.isNotBlank(auth.getServerId()) ?
```

Comment:
if `convertToAuthConfiguration` is only used in `login` method, shall we make it private?